### ISSUE TITLE
[feature request] Oric 1-Atmos firmware routines

### ISSUE THREAD

**Issue by Fabrizio-Caruso** [cc65/cc65#555]
@all but also specifically @polluks ,
it would be cool to have C functions to easily access the sound chip in the Oric 1 and Oric Atmos computer.

In particular it would be useful to have:
W8192
Possibly also
SOUND, MUSIC, PLAY (same as BASIC commands)

Remark: These rotines have different addresses on the Oric 1 and Oric Atmos.

I have attached a file with the description of the functions, their parameters and addresses on the two computer models.

![oricvsatmos](https://user-images.githubusercontent.com/29057615/34083421-098ed52c-e371-11e7-94a2-2069b4147db1.jpg)

**Comment by polluks** [comment]
Thank you so far. I prefer a system independent solution. Many years ago I already made a proposal.

**Comment by Fabrizio-Caruso** [comment]
@polluks, yes I do not mean to create a framework or full-fledged apis inside CC65.

But then if no APIs should be provided, why bother with the predefined sounds?

If providing many APIs is the problem, we could get rid of all of them, and only provide one single function to access the sound chip registers (W8192 in the doc, I have included).

This is sort of consistent with CC65 provides through pre-defined struct to access sound/graphics chips on system that allow direct memory-access.

**Comment by polluks** [comment]
I see, we already have 8912 support in https://github.com/cc65/cc65/blob/master/include/creativision.h.

**Comment by Fabrizio-Caruso** [comment]
@polluks, clearly I don't think removing the predefined sounds now is a good idea for backwards-compatibility.
I am only proposing to follow what is done on other targets and what seems to me the general rule in CC65: do not write entire machine-specific frameworks inside CC65 but provide accessors to chip registers. That is why I think the firmware routine W8192 (as in the attached document) should be made available.
On the other hand, SOUND, MUSIC and PLAY would be nice and useful extras, that might not comply with the rules. Personally, I would put then in, as well because they are almost no code.

**Comment by polluks** [comment]
Indeed, my "Tiny Sound Interface" shall not replace the build-in sounds. Every new API layer has stricter memory constraints but ROM routines are for free.
BTW it's not W8192 aka 0x2000 ;-)

**Comment by Fabrizio-Caruso** [comment]
@polluks, clearly I don't think removing the predefined sounds now is a good idea for backwards-compatibility.
I am only proposing to follow what is done on other targets and what seems to me the general rule in CC65: do not write entire machine-specific frameworks inside CC65 but provide accessors to chip registers. That is why I think the firmware routine W8192 (as in the attached document) should be made available.
On the other hand, SOUND, MUSIC and PLAY would be nice and useful extras, that might not comply with the rules. Personally, I would put then in, as well because they are almost no code.

**Comment by polluks** [comment]
@Fabrizio-Caruso Please check again!
